app/page.tsx

```diff
@@ -1,8 +1,17 @@
+'use client';
+
+import { useRouter } from "next/navigation";
+import { useEffect } from "react";
+
 const HomePage = () => {
+  const router = useRouter();
+
+  useEffect( () => {
+    router.push('/dashboard');
+  },[router]);
```

Comment:
Here we do not need to include the `router` as a dependency, we can keep it empty. So the `useEffect` will looks like as following:
```javascript
useEffect( () => {
  router.push('/dashboard');
} ,[]);
```